src/test/java/uk/gov/hmcts/ccd/domain/model/common/DisplayContextParameterUtilTest.java

```diff
@@ -0,0 +1,82 @@
+package uk.gov.hmcts.ccd.domain.model.common;
+
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import static java.util.Arrays.asList;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.junit.jupiter.api.Assertions.assertAll;
+import static uk.gov.hmcts.ccd.domain.model.common.DisplayContextParameterUtil.Parameter.COLLECTION;
+import static uk.gov.hmcts.ccd.domain.model.common.DisplayContextParameterUtil.Parameter.TABLE;
+import static uk.gov.hmcts.ccd.domain.model.common.DisplayContextParameterUtil.updateDisplayContextParameter;
+
+class DisplayContextParameterUtilTest {
+
+    @Test
+    @DisplayName("should create parameter)")
+    void shouldCreateParameter() {
+        assertAll(
+            () -> assertThat(updateDisplayContextParameter("", COLLECTION, asList("allowInsert", "allowDelete")),
+                is("#COLLECTION(allowDelete,allowInsert)")),
+            () -> assertThat(updateDisplayContextParameter("", COLLECTION, asList("allowDelete")),
+                is("#COLLECTION(allowDelete)")),
+            () -> assertThat(updateDisplayContextParameter("", TABLE, asList("postcode")),
+                is("#TABLE(postcode)")),
+            () -> assertThat(updateDisplayContextParameter("", TABLE, asList("AddressLine1", "postcode")),
+                is("#TABLE(AddressLine1,postcode)"))
+                 );
+    }
+
+    @Test
+    @DisplayName("should update parameter)")
+    void shouldUpdateParameter() {
+        assertAll(
+            () -> assertThat(updateDisplayContextParameter("#COLLECTION(allowInsert,allowDelete)", COLLECTION, asList("allowInsert", "allowDelete")),
+                is("#COLLECTION(allowDelete,allowInsert)")),
+            () -> assertThat(updateDisplayContextParameter("#COLLECTION(allowInsert,allowDelete)", COLLECTION, asList("allowDelete")),
+                is("#COLLECTION(allowDelete)")),
+            () -> assertThat(updateDisplayContextParameter("#COLLECTION()", COLLECTION, asList("allowInsert")),
+                is("#COLLECTION(allowInsert)")),
+            () -> assertThat(updateDisplayContextParameter("#TABLE(param1,param2)", TABLE, asList("postcode")),
```

Comment:
not sure this is correct. Data store should not interfere or override any TABLE option set on the definition. There's no reason to do that. I think we can restrict Data Store action only on the COLLECTION option and ignore all the rest